### PR TITLE
Add T&S pre-commit settings to ts_imsim.

### PR DESCRIPTION
**Pull request by jbkalmbach**

### CODE REVIEW

**Comment by gmegh** [comment]
Hi! Just jumping in here. Maybe we can wait to merge this until the other PR #14 is merged? I say so, because it seems this one applies flake8 and black to all the files, and it may be good to do it again with the updated code with snake_case.

File: .pre-commit-config.yaml
Comment (suberlak) [inline .pre-commit-config.yaml:1]: I notice it's called `.pre-commit-config.yaml` rather than `.ts_pre_commit_config.yaml`. Looking at https://tssw-developer.lsst.io/development-guidelines/tools/pre_commit.html#pre-commit  ,   is the former a more updated version ?

**Review by suberlak — APPROVED** [review]
Updates mirror the setup in `ts_wep`, in compliance with https://tssw-developer.lsst.io/development-guidelines/tools/pre_commit.html#pre-commit

File: .pre-commit-config.yaml
Comment (teweitsai) [reply .pre-commit-config.yaml:1]: We are using the `.ts_pre_commit_config.yaml` now.

Comment (jbkalmbach) [reply .pre-commit-config.yaml:1]: Hi @teweitsai, I had some issues with the way the yaml config construction works in `ts_imsim`. The only way to get the `check-yaml` pre-commit-hook to run correctly was to include a `.pre-commit-config.yaml` file generated by manually running the `ts-pre-commit-conf` package. See here: https://lsstc.slack.com/archives/CDS6B1LQP/p1698960417533709

Comment (jbkalmbach) [reply .pre-commit-config.yaml:1]: Looks like some of the behavior is not working as expected. I will hold off on this PR until the [`ts_pre_commit_conf` ](https://github.com/lsst-ts/ts_pre_commit_conf) behavior is fixed in that repository.

Comment (teweitsai) [reply .pre-commit-config.yaml:1]: It looks like @wvreeven is trying to fix it if I understand the conversation in Slack thread correctly. Thanks!

Comment (wvreeven) [reply .pre-commit-config.yaml:1]: I am indeed fixing this. PR submitted and in review.

Comment (teweitsai) [reply .pre-commit-config.yaml:1]: Got it. Thanks for the update!

**Review by teweitsai — APPROVED** [review]
Good to me

**Pull request merged**
